src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -201,8 +201,10 @@ public void Write(int pinNumber, PinValue value)
         /// <returns>A structure that contains the result of the waiting operation.</returns>
         public WaitForEventResult WaitForEvent(int pinNumber, PinEventTypes eventTypes, TimeSpan timeout)
         {
-            CancellationTokenSource tokenSource = new CancellationTokenSource(timeout);
-            return WaitForEvent(pinNumber, eventTypes, tokenSource.Token);
+            using (CancellationTokenSource tokenSource = new CancellationTokenSource(timeout))
+            {
+                return WaitForEvent(pinNumber, eventTypes, tokenSource.Token);
```

Comment:
If `WaitForEventResult` is awaitable, this could be disposing too early.

Reply:
True story, that is the case for the `WaitForEventAsync`, this one is fine because the `WaitForEventResult` is not awaitable

Reply:
I'm curious - will the timer still cancel even though the CTS is disposed? seems a bit confusing but I'm not an expert on this

Reply:
Disposing a CTS disposes of its Timer.